news/5380.feature

```diff
@@ -0,0 +1 @@
+This redefines the ``requests.Response.raise_for_status`` method with our own custom ``raise_for_status`` method
```

Comment:
This is too technical for the end user. I would say something like *refine error messages to avoid showing Python tracebacks when an HTTP error occurs*.